src/server/scripts/Northrend/IcecrownCitadel/boss_the_lich_king.cpp

```diff
@@ -514,6 +515,8 @@ class boss_the_lich_king : public CreatureScript
                 _vileSpiritExplosions = 0;
             }
 
+            void JustRespawned() override { }
```

Comment:
That's the point, Reset is called at https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Entities/Creature/Creature.cpp#L1718. If I do not override JustRespawned(), it will be called twice after a wipe